crates/common/src/lib.rs

```diff
@@ -74,3 +80,94 @@ pub fn to_wormhole_address(address: Address) -> B256 {
     bytes[12..].copy_from_slice(address.as_slice());
     B256::from(bytes)
 }
+
+/// Converts a Wormhole format B256 address to an Ethereum Address.
+pub fn from_wormhole_address_align(wormhole_addr: B256) -> Result<Address, String> {
+    // Extract the last 20 bytes from the 32-byte B256
+    // This reverses the Solidity conversion: bytes32(uint256(uint160(address)))
+    let bytes = wormhole_addr.as_slice();
+    let (garbage_bytes, addr_bytes) = bytes.split_at(12);
+
+    // Align 12 bytes into 4 byte chunks to create 3 u32 `ints`
+    let (prefix, ints, suffix) = unsafe { garbage_bytes.align_to::<u32>() };
+    if prefix.iter().any(|&b| b != 0)
+        || ints.iter().any(|&i| i != 0)
+        || suffix.iter().any(|&b| b != 0)
+    {
+        return Err(String::from("Malformed wormhole address"));
+    }
```

Comment:
This feels weird and not idiomatic to me. Not totally clear what this function is for